### PR TITLE
Remove deprecated BExecutor class usage

### PR DESCRIPTION
**Pull request by aashikam**
Fixes: https://github.com/ballerina-platform/module-ballerinax-kafka/issues/47

## Purpose
$title

### CODE REVIEW

**Review by Maninda — APPROVED** [review]

File: kafka-native/src/main/java/org/ballerinalang/messaging/kafka/impl/KafkaListenerImpl.java [hunk 1]
```diff
@@ -46,13 +45,13 @@
...
 
-    private Scheduler scheduler;
     private BObject service;
     private BObject listener;
     private ResponseCallback callback;
+    private BRuntime bRuntime;
 
-    public KafkaListenerImpl(Strand strand, BObject listener, BObject service) {
-        this.scheduler = strand.scheduler;
+    public KafkaListenerImpl(Strand strand, BObject listener, BObject service, BRuntime bRuntime) {
```
Comment (ThisaruGuruge) [inline kafka-native/src/main/java/org/ballerinalang/messaging/kafka/impl/KafkaListenerImpl.java:53]: Can we remove the `strand` now?

Comment (aashikam) [reply kafka-native/src/main/java/org/ballerinalang/messaging/kafka/impl/KafkaListenerImpl.java]: Yes. Fixed in f5efda80b5c70e350253b1fe80f70f7a79f9b9f7

File: kafka-native/src/main/java/module-info.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
     requires slf4j.api;
     requires java.transaction.xa;
     requires org.apache.avro;
+    requires com.fasterxml.jackson.core;
```
Comment (manuranga) [inline kafka-native/src/main/java/module-info.java:25]: accidentally added?

File: kafka-native/src/main/java/org/ballerinalang/messaging/kafka/impl/KafkaListenerImpl.java [hunk 3]
```diff
@@ -20,12 +20,11 @@
...
 import org.apache.kafka.clients.consumer.ConsumerRecords;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
-import org.ballerinalang.jvm.api.BExecutor;
+import org.ballerinalang.jvm.api.BRuntime;
 import org.ballerinalang.jvm.api.connector.CallableUnitCallback;
 import org.ballerinalang.jvm.api.values.BObject;
 import org.ballerinalang.jvm.observability.ObservabilityConstants;
 import org.ballerinalang.jvm.observability.ObserveUtils;
-import org.ballerinalang.jvm.scheduling.Scheduler;
 import org.ballerinalang.jvm.scheduling.Strand;
```
Comment (ThisaruGuruge) [inline kafka-native/src/main/java/org/ballerinalang/messaging/kafka/impl/KafkaListenerImpl.java:28]: Unused import? (Shall we add `checkstyle` and `spotbugs` to the Kafka module? Not with this PR, but later?)

File: kafka-native/src/main/java/module-info.java (hunk 2 above)
Comment (aashikam) [reply kafka-native/src/main/java/module-info.java:25]: Nope. I got.: 
```
> Task :kafka-native:compileJava FAILED
/Users/arshika/Developer/WSO2/Ballerina/Source/module-ballerinax-kafka/kafka-native/src/main/java/org/ballerinalang/messaging/kafka/nativeimpl/producer/SendAvroKeys.java:164: error: cannot access JsonParser
        Schema avroSchema = new Schema.Parser().parse(schemaString);
                                               ^
  class file for com.fasterxml.jackson.core.JsonParser not found
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 error
```

**Comment by manuranga** [comment]
looks good to me

File: kafka-native/src/main/java/org/ballerinalang/messaging/kafka/impl/KafkaListenerImpl.java (hunk 3 above)
Comment (aashikam) [reply kafka-native/src/main/java/org/ballerinalang/messaging/kafka/impl/KafkaListenerImpl.java:28]: Yes. https://github.com/ballerina-platform/module-ballerinax-kafka/issues/49

**Review by ThisaruGuruge — APPROVED** [review]
LGTM

**Pull request merged**